Gems/ROS2/Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp

```diff
@@ -74,15 +132,16 @@ namespace ROS2
         AZStd::vector<AZ::EntityId> createdEntities;
 
         AZStd::unordered_map<AZStd::string, AzToolsFramework::Prefab::PrefabEntityResult> createdLinks;
-        AzToolsFramework::Prefab::PrefabEntityResult createEntityRoot = AddEntitiesForLink(m_model->root_link_, AZ::EntityId(), createdEntities);
-        AZStd::string rootName(m_model->root_link_->name.c_str(), m_model->root_link_->name.size());
+        const sdf::Link* rootLink = GetFirstModel()->LinkByIndex(0);
+        AzToolsFramework::Prefab::PrefabEntityResult createEntityRoot = AddEntitiesForLink(rootLink, AZ::EntityId(), createdEntities);
+        AZStd::string rootName(GetFirstModel()->Name().c_str(), GetFirstModel()->Name().size());
         createdLinks[rootName] = createEntityRoot;
         if (!createEntityRoot.IsSuccess())
         {
             return AZ::Failure(AZStd::string(createEntityRoot.GetError()));
         }
 
-        auto links = Utils::GetAllLinks(m_model->root_link_->child_links);
+        auto links = Utils::GetAllLinks(*GetFirstModel(), true);
```

Comment:
Please correct me if I am wrong, but the following code will cause a duplication of the first link (the first link is already created in lines 135-136

![image](https://github.com/o3de/o3de-extras/assets/134940295/d4a115ce-f543-49d6-af89-cc71d61cdb81)

Reply:
Yes, this needs to be fixed.

There is no singular "root" link in an SDF file as it represents a graph structure which is different from URDF which is a tree structure where links the link hierarchy is established via the parent/child references in joints.

Reply:
The root link parsing logic has been fixed.
![image](https://github.com/o3de/o3de-extras/assets/56135373/7c00608f-b8e0-413c-be7f-dfe4b7412c6a)